src/attack_flow_builder/src/assets/builder.config.validator.ts

```diff
@@ -11,6 +11,10 @@ import {
 
 class AttackFlowValidator extends DiagramValidator {
 
+    static IPv4regex = /^(([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])\.){3}([0-9]|[1-9][0-9]|1[0-9]{2}|2[0-4][0-9]|25[0-5])(\/(3[0-2]|[1-2][0-9]|[0-9]))?$/;
+    static IPv6regex = /^s*((([0-9A-Fa-f]{1,4}:){7}([0-9A-Fa-f]{1,4}|:))|(([0-9A-Fa-f]{1,4}:){6}(:[0-9A-Fa-f]{1,4}|((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){5}(((:[0-9A-Fa-f]{1,4}){1,2})|:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3})|:))|(([0-9A-Fa-f]{1,4}:){4}(((:[0-9A-Fa-f]{1,4}){1,3})|((:[0-9A-Fa-f]{1,4})?:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){3}(((:[0-9A-Fa-f]{1,4}){1,4})|((:[0-9A-Fa-f]{1,4}){0,2}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){2}(((:[0-9A-Fa-f]{1,4}){1,5})|((:[0-9A-Fa-f]{1,4}){0,3}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(([0-9A-Fa-f]{1,4}:){1}(((:[0-9A-Fa-f]{1,4}){1,6})|((:[0-9A-Fa-f]{1,4}){0,4}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:))|(:(((:[0-9A-Fa-f]{1,4}){1,7})|((:[0-9A-Fa-f]{1,4}){0,5}:((25[0-5]|2[0-4]d|1dd|[1-9]?d)(.(25[0-5]|2[0-4]d|1dd|[1-9]?d)){3}))|:)))(%.+)?s*(\/(12[0-8]|1[0-1][0-9]|[1-9][0-9]|[0-9]))?$/;
```

Comment:
Did you mean `\s*` here? Currently you literally match 0 or more `s` at the beginning and end of an address.

Also, because the regex is case-insensitive, you can shorten the regex by removing the upper-case variants:
- `[0-9A-Fa-f]` -> `[0-9a-f]`

Reply:
Great catch, I'll push a fix.

Not sure if we want to update the corresponding regex in the stix observable file: https://github.com/center-for-threat-informed-defense/attack-flow/blob/f0f5f864e732a185fbf8815f492301bf3ade9472/stix/oasis-open/observables/ipv6-addr.json

What's odd to me is the coinciding ipv4 regex doesn't do this, but the ipv6 does.

Reply:
Fixed

Reply:
That's odd... Checking for spaces, especially at the beginning of a string, seems a bit unnecessary as you'd probably want to .trim() or .strip() the input anyway. The mistake seems to also be present here: https://github.com/oasis-open/cti-stix2-json-schemas/blob/master/schemas/observables/ipv6-addr.json

Definitely worth addressing at some point.